### PR TITLE
zpool: Don't print "repairing" on force faulted drives

### PR DESCRIPTION
**Pull request by tonyhutter**
### Motivation and Context
Fixes: #13927

### Description
If you force fault a drive that's resilvering/scrubbing, it's scan stats can get frozen in time, giving the false impression that it's being resilvered.  This commit checks the vdev state to see if the vdev is healthy before reporting "resilvering" or "repairing" in zpool status.

### How Has This Been Tested?
We had a vdev that was force-faulted while either resilvering or reparing from a scrub.  It reported this to zpool status:
```
	    U78     FAULTED      3     2     0  external device fault  (repairing)
```
This was because the faulted U78 had a non-zero `vs->vs_scan_processed` and assumed it was being repaired.  However, added debug prints showed that its `vs->vs_scan_processed` value was not increasing, and this was not actually repairing while faulted.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: cmd/zpool/zpool_main.c [hunk 1]
```diff
@@ -2462,7 +2462,14 @@ print_status_config(zpool_handle_t *zhp, status_cbdata_t *cb, const char *name,
...
 
-	if (ps != NULL && ps->pss_state == DSS_SCANNING && children == 0) {
+	/*
+	 * If you force fault a drive that's resilvering, it's scan stats can
+	 * get frozen in time, giving the false impression that it's
+	 * being resilvered.  That's why check the state to see if the vdev
+	 * is healthy before reporting "resilvering" or "repairing".
+	 */
+	if (ps != NULL && ps->pss_state == DSS_SCANNING && children == 0 &&
+	    vs->vs_state == VDEV_STATE_HEALTHY) {
```
Comment (behlendorf) [inline cmd/zpool/zpool_main.c:2472]: You'll want to add the same check for `vs->vs_rebuild_processed` below.  This is the counter which is incremented for the sequential resilver, `zpool replace -s`.

File: cmd/zpool/zpool_main.c [hunk 2]
```diff
@@ -2462,7 +2462,14 @@ print_status_config(zpool_handle_t *zhp, status_cbdata_t *cb, const char *name,
 	(void) nvlist_lookup_uint64_array(root, ZPOOL_CONFIG_SCAN_STATS,
 	    (uint64_t **)&ps, &c);
 
-	if (ps != NULL && ps->pss_state == DSS_SCANNING && children == 0) {
+	/*
+	 * If you force fault a drive that's resilvering, it's scan stats can
```
Comment (behlendorf) [inline cmd/zpool/zpool_main.c:2466]:
```suggestion
	 * If you force fault a drive that's resilvering, its scan stats can
```

File: cmd/zpool/zpool_main.c [hunk 3]
```diff
@@ -2462,7 +2462,14 @@ print_status_config(zpool_handle_t *zhp, status_cbdata_t *cb, const char *name,
...
+	 * get frozen in time, giving the false impression that it's
+	 * being resilvered.  That's why check the state to see if the vdev
```
Comment (behlendorf) [inline cmd/zpool/zpool_main.c:2468]:
```suggestion
	 * being resilvered.  That's why we check the state to see if the vdev
```

**Comment by tonyhutter** [comment]
@behlendorf thanks, all fixed in my latest push.

**Review by behlendorf — APPROVED** [review]

**Comment by tonyhutter** [comment]
My latest push fixes the 72-char commit message checkstyle error.  Should be good to go

**Pull request merged**